app/controllers/likes_controller.rb

```diff
@@ -0,0 +1,42 @@
+class LikesController < ApplicationController
+  def new
+    @like = Like.new
+    @user = current_user
+    @post = Post.first
+    respond_to do |format|
+      format.html { render :new, locals: { like: @like } }
+    end
+  end
+
+  def create
+    # new object from params
+    @like = Like.new(post_params)
+
+    # respond_to block
+    respond_to do |format|
+      format.html do
+        if @like.save
+          # success message
+          post = Post.find(post_params[:author_id])
+          Like.update_post_likes_count(post)
+          flash[:success] = 'Liked'
+          # redirect to index
+          redirect_to user_posts_url
+        else
+          flash.now[:error] = 'Error: Like could not be added'
+          # render new
+          render :new, locals: { like: @like }
+        end
+      end
+    end
+  end
```

Comment:
#### Great work configuring routes for the Like model. However, there are some issues I noticed when I try to like a post:
- [ ] When I click the Add Like button, it takes me to a particular page that requests a label attribute. If you check your `schema.rb` file, you will see that the like table has no column for a label. and when I feel the form, it doesn't add a like to the required post. To fix this, I would advise you to do the following:

- Move the **_Add Like_** button to the `post#show` page so that you can like that particular post.
- Find the particular post in the controller by the id of the post passed in as params and then assign the id of the post to post_params[:post_id].
- Have a button in the `post#show` page that links to the create method in the LikesController. ` <%= link_to like_comment_url, method: :post %>`

Reply:
I had to add the label because I was getting an error with a field that why I added a placeholder one

Reply:
Hi @jssol, I understand but one of the project requirements is to follow the schema diagram for this project. Kindly implement the changes and you will be fine. 👍